src/Layer/C3DTilesLayer.js

```diff
@@ -142,48 +409,152 @@ class C3DTilesLayer extends GeometryLayer {
+        tileContent.traverse((child) => {
+            if (child.geometry && child.geometry.attributes._BATCHID) {
+                this.tilesC3DTileFeatures.set(tileContent.tileId, new Map());// initialize
+                const batchTable = this.findBatchTable(child);
+                if (!batchTable) {
+                    throw new Error('no batchTable');
+                }
+
+                const geoAttributes = child.geometry.attributes;
+                let currentBatchId = geoAttributes._BATCHID.array[0];
+                let start = 0;
+                let count = 0;
+
+                const registerBatchIdGroup = () => {
+                    if (this.tilesC3DTileFeatures.get(tileContent.tileId).has(currentBatchId)) {
+                        // already created
+                        const c3DTileFeature = this.tilesC3DTileFeatures.get(tileContent.tileId).get(currentBatchId);
+                        // add new group
+                        c3DTileFeature._groups.push({
+                            start,
+                            count,
+                        });
+                    } else {
+                        // first occurence
+                        const c3DTileFeature = new C3DTileFeature(
+                            this,
+                            tileContent.tileId,
+                            currentBatchId,
+                            [{ start, count }], // initialize with current group
+                            batchTable.getInfoById(currentBatchId),
+                        );
+                        this.tilesC3DTileFeatures.get(tileContent.tileId).set(currentBatchId, c3DTileFeature);
+                    }
+                };
+
+                for (let index = 0; index < geoAttributes.position.array.length; index += geoAttributes.position.itemSize) {
+                    const batchIndex = index / geoAttributes.position.itemSize;
+                    const batchId = geoAttributes._BATCHID.array[batchIndex];
+
+                    // check if batchId is currentBatchId
+                    if (currentBatchId !== batchId) {
+                        registerBatchIdGroup();
+
+                        // reset
+                        currentBatchId = batchId;
+                        start = batchIndex;
+                        count = 0;
+                    }
+
+                    // record this position in current C3DTileFeature
+                    count++;
+
+                    // check if end of the array
+                    if (index + geoAttributes.position.itemSize >= geoAttributes.position.array.length) {
+                        registerBatchIdGroup();
+                    }
+                }
+            }
+        });
+
+        this._styleController.updateStyle(this, [tileContent.tileId]);// only update tile batchelement
+    }
+
+    /**
+     * Explicit tell the style controller to update style (could be call when Style function have changed)
```

Comment:
`Could be call when Style function have changed` -> to remove ?

Reply:
Actually this is the only case where you have to call this method since C3DTileFeature modification handle style update

Reply:
with the new behavior ie no auto update for the feature, this should be call everytime style should be updated ie (style function have changed or feature have changed)